### PR TITLE
feat: [SRM-15071]: enabling validations for analyze step

### PR DESCRIPTION
**Pull request by harshilbits**
### Summary
1. Enabling validations for analyze step

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/85-cv/components/PipelineSteps/AnalyzeDeploymentImpact/components/AnalyzeDeploymentImpactWidget/AnalyzeDeploymentImpactWidget.tsx [hunk 1]
```diff
@@ -50,15 +44,14 @@ export function AnalyzeDeploymentImpactWidget(
...
-        // This will be added in the next PR.
-        // const errors = validateForm(data)
-        // if (!isEmpty(errors)) {
-        //   onChange?.(data)
-        //   return errors
-        // } else {
-        //   onChange?.(data)
-        // }
+      validate={data => {
+        const errors = validateForm(data)
```
Comment (karansaraswat19) [inline src/modules/85-cv/components/PipelineSteps/AnalyzeDeploymentImpact/components/AnalyzeDeploymentImpactWidget/AnalyzeDeploymentImpactWidget.tsx:48]: Can we do something like here?
```
const errors = validateForm(data)
onChange?.(data)
if (!isEmpty(errors)) {
   return errors
}
```

Comment (harshilbits) [reply src/modules/85-cv/components/PipelineSteps/AnalyzeDeploymentImpact/components/AnalyzeDeploymentImpactWidget/AnalyzeDeploymentImpactWidget.tsx:48]: done.

**Review by karansaraswat19 — APPROVED** [review]

File: src/modules/85-cv/components/PipelineSteps/AnalyzeDeploymentImpact/components/AnalyzeDeploymentImpactWidget/AnalyzeDeploymentImpactWidget.tsx [hunk 2]
```diff
@@ -50,15 +44,12 @@ export function AnalyzeDeploymentImpactWidget(
...
-      validate={() => {
-        // This will be added in the next PR.
-        // const errors = validateForm(data)
-        // if (!isEmpty(errors)) {
-        //   onChange?.(data)
-        //   return errors
-        // } else {
-        //   onChange?.(data)
-        // }
+      validate={data => {
```
Comment (chetanvekariya-harness) [inline src/modules/85-cv/components/PipelineSteps/AnalyzeDeploymentImpact/components/AnalyzeDeploymentImpactWidget/AnalyzeDeploymentImpactWidget.tsx:47]: Can't we use validation schema instead?

**Review by chetanvekariya-harness — APPROVED** [review]

File: src/modules/85-cv/components/PipelineSteps/AnalyzeDeploymentImpact/components/AnalyzeDeploymentImpactWidget/AnalyzeDeploymentImpactWidget.tsx (hunk 2 above)
Comment (harshilbits) [reply src/modules/85-cv/components/PipelineSteps/AnalyzeDeploymentImpact/components/AnalyzeDeploymentImpactWidget/AnalyzeDeploymentImpactWidget.tsx:47]: These are complex nested validations with different conditions , easier to manage and handle them by defining them in function.

**Pull request merged**